47.md

```diff
@@ -126,6 +126,189 @@ Response:
 Errors:
 - `PAYMENT_FAILED`: The payment failed. This may be due to a timeout, exhausting all routes, insufficient capacity or similar.
 
+### `pay_keysend`
+
+Request:
+```jsonc
+{
+    "method": "pay_keysend",
+    "params": {
+        "amount": 123, // invoice amount in msats, required
+        "pubkey": "03...", // payee pubkey, required
+        "preimage": "0123456789abcdef...", // preimage of the payment, optional
+        "tlv_records: [ // tlv records, optional
+            {
+                "type": 5482373484, // tlv type
+                "value": "fajsn341414fq" // hex encoded tlv value
+            }
+        ]
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "pay_keysend",
+    "result": { // result, object. null in case of error.
+        "preimage": "0123456789abcdef...", // preimage of the payment
+    }
+}
+```
+
+### `make_invoice`
+
+Request:
+```jsonc
+{
+    "method": "make_invoice",
+    "params": {
+        "amount": 123, // invoice amount in msats, required
+        "description": "some description", // invoice description, optional
+        "description_hash": "31afdf1..", // invoice description hash, optional
+        "preimage": "0123456789abcdef...", // preimage for the payment, optional
```

Comment:
@benthecarman @bumi I think there are two cases for making invoices:

1. You create a connection string with only make_invoice permissions and give it to someone else. Then they can pay you. (like LNURL)
2. You create a service that has access to a connection string with make_invoice permissions to receive payments on behalf of users of the service.

Is there a good usecase for being able to set a custom preimage? because I think it breaks the first case.

Reply:
Yeah I guess you might not anyone with a NWC connection to be able to create with custom preimages. Ideally we don't remove the functionality and this is enforced client side, however, that makes permissioning harder since normally it just around the commands. We could add another command but that isn't very pretty.

Reply:
Going to remove for now. Since it is an added field this can be added in the future without breaking backwards compatibility

Reply:
Sounds good to me, thanks!